core/operation/src/main/java/uk/gov/gchq/gaffer/operation/data/EdgeSeed.java

```diff
@@ -56,58 +77,76 @@ public Object getSource() {
     }
 
     @Override
-    public void setSource(final Object source) {
-        this.source = source;
+    public Object getDestination() {
+        return destination;
     }
 
     @Override
-    public Object getDestination() {
-        return destination;
+    public DirectedType getDirectedType() {
+        return directed;
     }
 
     @Override
-    public void setDestination(final Object destination) {
+    public void setIdentifiers(final Object source, final Object destination, final DirectedType directed, final MatchedVertex matchedVertex) {
+        this.source = source;
         this.destination = destination;
+        this.directed = directed;
+        this.matchedVertex = matchedVertex;
+        orderVertices();
     }
 
     @Override
-    public DirectedType getDirectedType() {
-        return directed;
+    public MatchedVertex getMatchedVertex() {
+        return matchedVertex;
     }
 
-    @Override
-    public void setDirectedType(final DirectedType directed) {
-        if (null == directed) {
-            this.directed = DirectedType.EITHER;
-        } else {
-            this.directed = directed;
+    private void orderVertices() {
```

Comment:
Could this method be moved to EdgeId to reduce duplication? It's shared between the EdgeSeed and Edge classes.